R/utils_coverage.R

```diff
@@ -0,0 +1,230 @@
+                         vacc_date_col,
+                         unit = c("day", "month", "year"),
+                         date_interval = NULL) {
+  checkmate::assert_data_frame(
+    data_set,
+    min.rows = 1L
+  )
+  checkmate::assert_character(
+    vacc_date_col,
+    min.len = 1L
+  )
+  checkmate::assert_names(
+    colnames(data_set),
+    must.include = c(vacc_date_col)
+  )
+  checkmate::assert_date(
+    data_set[[vacc_date_col]]
+  )
+  unit <- match.arg(unit, several.ok = FALSE)
+  checkmate::assert_string(
+    unit
+  )
+
+  # Create continuous date column
+  # For fixed intervals, use date_intervales
+  # In other case use min and max of data_set
+  if (is.null(date_interval)) {
+    start <- min(data_set[[vacc_date_col]], na.rm = TRUE)
+    end <- max(data_set[[vacc_date_col]], na.rm = TRUE)
+  } else {
+    checkmate::assert_date(
+      date_interval
+    )
+    start <- date_interval[1]
+    end <- date_interval[2]
+  }
+  start <- as.Date(trunc(as.POSIXlt(start), units = unit))
+  dates <- seq(from = start, to = end, by = unit)
+  coverage <- data.frame(date  = dates)
+
+  count <- as.data.frame(
+    table(
+      cut(data_set[[vacc_date_col]], breaks = unit),
+      dnn = "date"
+    ),
+    responseName = "doses"
+  )
+  count$date <- as.Date(count$date)
+  coverage <- merge(x = coverage, y = count,
+    by = "date", all.x = TRUE
+  )
+  coverage$doses[is.na(coverage$doses)] <- 0
+  coverage$cum_doses <- cumsum(coverage$doses)
+  coverage$coverage <- coverage$cum_doses / nrow(data_set)
+  return(coverage)
+}
+
+#' @title Plot Vaccine Coverage
+#'
+#' @description This function returns a plot of the vaccine coverage or the
+#' cumulative coverage (if cumulative = TRUE). The return is a 2-axis `ggplot2`
+#' element with the number of vaccines per date on the left axis and the
+#' coverage per date on the right axis. When a matching routine is performed,
+#' the left axis also accounts for the doses of the matched cohort.
+#' @inheritParams coh_coverage
+#' @param vaccineff_data Object of the class `vaccineff_data` with
+#' vaccineff data.
+#' @param cumulative If `TRUE`, returns the cumulative number of doses over the
+#' time window.
+#' @return 2-axis ggplot2 plot of vaccine coverage and daily doses.
+#' @examples
+#' # Load example data
+#' data("cohortdata")
+#'
+#' # Create `vaccineff_data`
+#' vaccineff_data <- make_vaccineff_data(data_set = cohortdata,
+#'   outcome_date_col = "death_date",
+#'   censoring_date_col = "death_other_causes",
+#'   vacc_date_col = "vaccine_date_2",
+#'   vaccinated_status = "v",
+#'   unvaccinated_status = "u",
+#'   immunization_delay = 15,
+#'   end_cohort = as.Date("2044-12-31"),
+#'   match = TRUE,
+#'   exact = c("age", "sex"),
+#'   nearest = NULL
+#' )
+#' plot_coverage(vaccineff_data = vaccineff_data,
+#'   date_interval = as.Date(c("2044-01-03", "2044-09-30")),
+#'   cumulative = FALSE
+#' )
+#' @export
+
+plot_coverage <- function(vaccineff_data,
```

Comment:
Could `plot_coverage()` become a `plot` method for the `<vaccineff_data>` class?

Reply:
Good suggestions. I will address it in https://github.com/epiverse-trace/vaccineff/issues/85